pkg/local_object_storage/blobstor/common/get.go

```diff
@@ -15,3 +15,11 @@ type GetRes struct {
 	Object  *objectSDK.Object
 	RawData []byte
 }
+
+// GetBytesOptions groups options for [Storage.GetBytes].
+type GetBytesOptions struct {
+	// GetBuffer specifies function replacing built-in make to allocate memory
+	// buffer of ln length. Buffer allocated via GetBuffer is returned from GetBytes
+	// regardless of error so caller could handle it properly.
+	GetBuffer func(ln int) []byte
```

Comment:
I'd rather have `buf []byte` passed. In most cases we know the size already.

Reply:
we never know it, only inside end storage (fstree, peapod)

Reply:
Meta has payload size, we request things after request to metabase IINM.

Reply:
yes, but here `ln` stand for complete binary object, not only payload. Dynamic size resolution is easier and more reliable (blob storage could diverge with metabase in theory)

at the same time, we _can_ add API to request full binary size from metabase, but in my opinion it's not worth it for this particular goal

Reply:
gone